ml/src/main/scala/org/apache/spark/ml/h2o/param/H2OAlgoParamsHelper.scala

```diff
@@ -133,4 +129,13 @@ trait H2OAlgoParamsHelper[P <: Parameters] extends Params {
   protected def nullableStringArrayParam(name: String, doc: Option[String] = None): NullableStringArrayParam = {
     new NullableStringArrayParam(this, name, getDoc(doc, name))
   }
+
+  protected def checkAllowedEnumValues[T <: Enum[T]](name: String, nullAllowed: Boolean = false)
+                                                    (implicit ctag: reflect.ClassTag[T]): Unit = {
+    val names = ctag.runtimeClass.getDeclaredMethod("values").invoke(null).asInstanceOf[Array[T]].map(_.name())
+    if (name == null && !nullAllowed || !names.contains(name)) {
```

Comment:
Wouldn't be more convenient to be case insensitive?

Reply:
Good point, we had that on Python side as well, will apply that on Scala side as well

Reply:
Hm, more I think about it, it seems better to me to be strict about this. I would actually prefer to be case senstive.